jumpscale/packages/vdc_dashboard/sals/vdc_dashboard_sals.py

```diff
@@ -241,3 +241,17 @@ def _get_ingress_used_ports():
                 ports.add(path["backend"]["servicePort"])
 
     return ports
+
+
+def get_ingresstcp_used_ports():
+    tcp_ports = set()
+    config_path = j.sals.fs.expanduser("~/.kube/config")
+    k8s_client = j.sals.kubernetes.Manager(config_path=config_path)
+    out = k8s_client.execute_native_cmd("kubectl get IngressRouteTCP -A -o json")
```

Comment:
this line should be surrounded with try/except for `j.exceptions.Runtime` which may happen and propagate a readable error to the user

Reply:
OK , lets add that too then

Reply:
Should be good with https://github.com/threefoldtech/js-sdk/pull/3082/commits/045ce8e9593938630fc9f952e796480bafae75e8